src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaInstructorNoteService.java

```diff
@@ -0,0 +1,60 @@
+package edu.ucdavis.dss.ipa.services.jpa;
+
+import edu.ucdavis.dss.ipa.entities.Instructor;
+import edu.ucdavis.dss.ipa.entities.InstructorNote;
+import edu.ucdavis.dss.ipa.entities.Schedule;
+import edu.ucdavis.dss.ipa.repositories.InstructorNoteRepository;
+import edu.ucdavis.dss.ipa.repositories.InstructorRepository;
+import edu.ucdavis.dss.ipa.repositories.ScheduleRepository;
+import edu.ucdavis.dss.ipa.services.InstructorNoteService;
+import edu.ucdavis.dss.ipa.utilities.EmailService;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+
+@Service
+public class JpaInstructorNoteService implements InstructorNoteService {
+  @Inject InstructorNoteRepository instructorNoteRepository;
+  @Inject ScheduleRepository scheduleRepository;
+  @Inject InstructorRepository instructorRepository;
+
+  /**
+   * Will either create the instructorNote, or find one that matches on schedule and instructor.
+   * @param scheduleId
+   * @param instructorId
+   * @return
+   */
+  @Override
+  public InstructorNote findOrCreateByScheduleIdAndInstructorId(long scheduleId, long instructorId) {
+    InstructorNote instructorNote = instructorNoteRepository.findByScheduleIdAndInstructorId(scheduleId, instructorId);
+
+    if (instructorNote == null) {
+      instructorNote = this.create(scheduleId, instructorId);
+    }
+
+    return instructorNote;
+  }
+
+  private InstructorNote create(long scheduleId, long instructorId) {
+    InstructorNote instructorNote = new InstructorNote();
+    Schedule schedule = scheduleRepository.findOne(scheduleId);
```

Comment:
No need for the temporary variables 'schedule' and 'instructor' if you never check them and only use them once.